wremnants/syst_tools.py

```diff
@@ -0,0 +1,21 @@
+import hist
+import numpy as np
+
+def scale_helicity_hist_to_variations(scale_hist):
+
+    s = hist.tag.Slicer()
+    # select nominal QCD scales, but keep the sliced axis at size 1 for broadcasting
+    nom_scale_hist = scale_hist[{"muRfact" : s[1.j:1.j+1], "muFfact" : s[1.j:1.j+1]}]
+
+    # select nominal QCD scales and project down to nominal axes
+    nom_hist = nom_scale_hist[{"ptVgen" : s[::hist.sum], "chargeVgen" : s[::hist.sum], "helicity" : s[::hist.sum], "muRfact" : s[1.j], "muFfact" : s[1.j] }]
+
+
+
+    # difference between a given scale and the nominal, plus the sum
+    # this emulates the "weight if idx else nominal" logic and corresponds to the decorrelated
+    # variations
+    scale_variation_hist = hist.Hist(*scale_hist.axes, storage = scale_hist._storage_type(), name = scale_hist.name + "_variations",
```

Comment:
If this hist doesn't have a name (as is the case in what I'm testing, not sure if that's on my side or not), this fails

Reply:
Added a protection and default name.